js/viz/axes/base_axis.js

```diff
@@ -1097,44 +1025,202 @@ Axis.prototype = {
 
-        that._majorTicks = that._minorTicks = null;
-        that._updateTickManager();
+        boundaryTicks = that._getBoundaryTicks(ticks.ticks);
+        if(options.showCustomBoundaryTicks && boundaryTicks.length) {
+            that._boundaryTicks = [boundaryTicks[0]].map(createBoundaryTick(that, renderer, true));
+            if(boundaryTicks.length > 1) {
+                that._boundaryTicks = that._boundaryTicks.concat([boundaryTicks[1]].map(createBoundaryTick(that, renderer, false)));
+            }
+        }
+
+        var minors = (ticks.minorTicks || []).filter(function(minor) {
+            return !boundaryTicks.some(function(boundary) {
+                return valueOf(boundary) === valueOf(minor);
+            });
+        });
 
-        that._majorTicks = tickManager.getTicks().map(createMajorTick(this, renderer));
-        that._minorTicks = tickManager.getMinorTicks().map(createMinorTick(this, renderer));
+        that._tickInterval = ticks.tickInterval;
+        that._minorTickInterval = ticks.minorTickInterval;
+
+        that._majorTicks = ticks.ticks.map(createMajorTick(that, renderer, that._getSkippedCategory(ticks.ticks)));
+        that._minorTicks = minors.map(createMinorTick(that, renderer));
 
         that.correctTicksOnDeprecated();
 
-        boundaryTicks = that._getBoundaryTicks();
-        if(this._options.showCustomBoundaryTicks && boundaryTicks.length) {
-            that._boundaryTicks = [boundaryTicks[0]].map(createBoundaryTick(this, renderer, true));
-            if(boundaryTicks.length > 1) {
-                that._boundaryTicks = that._boundaryTicks.concat([boundaryTicks[1]].map(createBoundaryTick(this, renderer, false)));
+        that.reinitTranslator();
+    },
+
+    reinitTranslator: function() {
+        var that = this,
+            range = that._getViewportRange(),
+            minVisible = range.minVisible,
+            maxVisible = range.maxVisible,
+            interval = range.interval,
+            ticks = that._majorTicks,
+            length = ticks.length;
+
+        if(that._options.type !== constants.discrete) {
+            if(!range.isSynchronized &&
+                length &&
+                !that._options.skipViewportExtending &&
+                (!isDefined(that._zoomArgs) || !that.isArgumentAxis)) {
+                if(ticks[0].value < range.minVisible) {
+                    minVisible = ticks[0].value;
+                }
+                if(length > 1 && ticks[length - 1].value > range.maxVisible) {
+                    maxVisible = ticks[length - 1].value;
+                }
             }
+
+            interval = this._calculateRangeInterval(that.calculateInterval(maxVisible, minVisible), interval);
+
+            range.addRange({
+                minVisible: minVisible,
+                maxVisible: maxVisible,
+                interval: interval
+            });
         }
+        that._translator.updateBusinessRange(range);
+    },
+
+    //for internal use only
```

Comment:
The underscore at the beginning of the function name indicates 'for internal use only', isn't it?